runtime-modules/forum/src/lib.rs

```diff
@@ -2342,4 +2348,12 @@ impl<T: Config> Module<T> {
             Err(Error::<T>::DataMigrationNotDone)
         }
     }
+
+    fn ensure_empty_thread(thread: &ThreadOf<T>) -> DispatchResult {
+        ensure!(
+            thread.number_of_posts == 0u64,
```

Comment:
```suggestion
            thread.number_of_posts.is_zero(),
```
In case we decide to change the type